server/src/main/java/org/elasticsearch/index/IndexMode.java

```diff
@@ -273,9 +273,16 @@ public String getName() {
     @Nullable
     public abstract CompressedXContent getDefaultMapping();
 
+    /**
+     * Build the {@link FieldMapper} for {@code _id}.
+     */
     public abstract IdFieldMapper buildIdFieldMapper(BooleanSupplier fieldDataEnabled);
 
-    public abstract IdFieldMapper buildNoFieldDataIdFieldMapper();
+    /**
+     * Get the singleton {@link FieldMapper} for {@code _id}. It can never support
+     * field data.
+     */
+    public abstract IdFieldMapper idFieldMapperWithoutFieldData();
```

Comment:
This may be a dumb question, but is there a reason we can't just call `buildIdfieldMapper(() -> false)` here instead?

Reply:
I kind of like having a version that never builds things - just returns singletons. And I like having the behavior explicit in the name. Basically - I like it this was better :)